### PR TITLE
Feature 34 survey publication

### LINKED ISSUE #39: Publish survey
As a user I'd like to be able to publish a survey

### PR DESCRIPTION
**Pull request by villejuutila**
Resolves #39

### CODE REVIEW

File: src/components/editor/devices-panel.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    if (!foundDevice?.id) return;
+
+    if (checked) {
+      setSelectedDevices([...selectedDevices, foundDevice]);
+    } else {
+      setSelectedDevices(selectedDevices.filter((device) => device.id !== foundDevice.id));
+    }
+  };
+  /**
```
Comment (EVP1989) [inline src/components/editor/devices-panel.tsx:35]:
```suggestion
    }
  };

  /**
```

File: src/components/editor/publish-survey.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          ? DeviceSurveyStatus.Scheduled
+          : DeviceSurveyStatus.Published,
+      publishStartTime: publishStartTime,
+      publishEndTime: publishEndTime
+    }));
+
+    await publishSurveys(deviceSurveys);
+  };
+  return (
+    <>
```
Comment (EVP1989) [inline src/components/editor/publish-survey.tsx:47]:
```suggestion
  };

  return (
    <>
```

File: src/components/editor/publish-survey.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+   * Publishes survey
+   *
+   * @param publishStartTime publish start time
+   * @param publishEndTime publish end time
+   */
+  const handlePublish = async (publishStartTime?: Date, publishEndTime?: Date) => {
+    if (!survey.id) return;
+    const deviceSurveys: DeviceSurvey[] = selectedDevices.map((device) => ({
+      deviceId: device.id!,
+      surveyId: survey.id!,
```
Comment (EVP1989) [inline src/components/editor/publish-survey.tsx:35]:
```suggestion
      surveyId: survey.id,
```

**Review by EVP1989 — APPROVED** [review]
Approved with few suggestions

File: src/components/editor/devices-panel.tsx (hunk 1 above)
Comment (villejuutila) [reply src/components/editor/devices-panel.tsx:35]: Resolved.

File: src/components/editor/publish-survey.tsx (hunk 3 above)
Comment (villejuutila) [reply src/components/editor/publish-survey.tsx:35]: Typescript doesn't seem to realize that it cannot be null so build fails without the non-null assertion.

Comment (EVP1989) [reply src/components/editor/publish-survey.tsx:35]: Mysterious, well let's leave it as it is then

**Pull request merged**